server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4451,18 +4460,14 @@ public void finalizeStop(VirtualMachineProfile profile, Answer answer) {
             throw new PermissionDeniedException("The owner of " + vm + " is disabled: " + vm.getAccountId());
         }
 
-        Host destinationHost = null;
-        if (hostId != null) {
-            Account account = CallContext.current().getCallingAccount();
-            if (!_accountService.isRootAdmin(account.getId())) {
-                throw new PermissionDeniedException(
-                        "Parameter hostid can only be specified by a Root Admin, permission denied");
-            }
-            destinationHost = _hostDao.findById(hostId);
-            if (destinationHost == null) {
-                throw new InvalidParameterValueException("Unable to find the host to deploy the VM, host id=" + hostId);
-            }
-        }
+        Account account = CallContext.current().getCallingAccount();
+        boolean isRootAdmin = _accountService.isRootAdmin(account.getId());
+
+        Pod destinationPod = getDestinationPod(podId, isRootAdmin);
```

Comment:
what if,
the user passed a pod, ...

Reply:
pod will be chosen for the deployment if cluster and host have not been specified.